AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/PinpointContext.swift

```diff
@@ -19,17 +61,132 @@ class PinpointContext {
         EndpointClient(context: self)
     }()
     
-    lazy var sessionTracker: SessionTracker = {
-        SessionTracker(context: self)
+    lazy var sessionTracker: SessionClient = {
+        SessionClient(context: self)
     }()
     
-    init(with configuration: PinpointClient.PinpointClientConfiguration) {
-        pinpointClient = PinpointClient(config: configuration)
+    private let keychainStore: KeychainStoreBehavior
+
+    init(with configuration: PinpointContextConfiguration,
+         credentialsProvider: CredentialsProvider,
+         region: String,
+         userDefaults: UserDefaultsBehaviour = UserDefaults.standard,
+         keychainStore: KeychainStoreBehavior = KeychainStore(service: Constants.Keychain.service)) throws {
+        self.configuration = configuration
+        self.keychainStore = keychainStore
+        self.userDefaults = userDefaults
+        let pinpointConfiguration = try PinpointClient.PinpointClientConfiguration(credentialsProvider: credentialsProvider,
+                                                                                   frameworkMetadata: AmplifyAWSServiceConfiguration.frameworkMetaData(),
+                                                                                   region: region)
+        pinpointClient = PinpointClient(config: pinpointConfiguration)
     }
     
-    var legacyUniqueId: String {
-        // TODO: Implement
-        fatalError("Not yet implemented")
+    private var legacyPreferencesFilePath: String? {
+        let applicationSupportDirectoryUrls = FileManager.default.urls(for: .applicationSupportDirectory,
+                                                                      in: .userDomainMask)
+        let preferencesFileUrl = applicationSupportDirectoryUrls.first?
+            .appendingPathComponent(Constants.Preferences.mobileAnalyticsRoot)
+            .appendingPathComponent(configuration.appId)
+            .appendingPathComponent(Constants.Preferences.fileName)
+        
+        return preferencesFileUrl?.path
+    }
+    
+    private func removeLegacyPreferencesFile() {
+        guard let preferencesPath = legacyPreferencesFilePath else {
+            return
+        }
+        
+        do {
+            try FileManager.default.removeItem(atPath: preferencesPath)
```

Comment:
in order to fully test this (and the `legacyUniqueId` and `retrieveUniqueId` methods), we should also inject an abstraction of FileManager in the initializer
We can perhaps have a protocol similar https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyTools/AmplifyXcode/Sources/AmplifyXcodeCore/Environment/AmplifyFileManager.swift
and then just add conformance to Foundation `FileManager`

Reply:
Yep, I definitely agree. I'll add it